packages/eslint-plugin-testing-library-react/src/rules/prefer-expect-query-by.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import { createRule } from '../utils';
+
+interface ExtendedExpression extends TSESTree.CallExpression {
+  callee: TSESTree.MemberExpression | TSESTree.Identifier;
+}
+
+function isIdentifier(
+  node: ExtendedExpression['callee']
+): node is TSESTree.Identifier {
+  return node.type === AST_NODE_TYPES.Identifier;
+}
+
+function isMemberExpression(
+  node: ExtendedExpression['callee']
+): node is TSESTree.MemberExpression {
+  return node.type === AST_NODE_TYPES.MemberExpression;
+}
+
+function isUsingWrongQueries(node: TSESTree.Identifier) {
+  return node.name.startsWith('getBy') || node.name.startsWith('getAllBy');
+}
+
+// https://github.com/microsoft/TypeScript/issues/16069#issuecomment-369374214
+function isNotNullOrUndefined<T>(input: null | undefined | T): input is T {
+  return input != null;
+}
+
+function mapNodesForWrongGetByQuery(node: TSESTree.CallExpression) {
```

Comment:
Relating to the comment above this is just finding the candidates, right?

Reply:
Yes, the nodes that refer to the `getBy` and `getAllBy`, which we also use for fixing the rule.